resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -142,22 +150,31 @@ private[yarn] class YarnAllocator(
   } else {
     0
   }
-  // Number of cores per executor.
-  protected val executorCores = sparkConf.get(EXECUTOR_CORES)
+  // Number of cores per executor for the default profile
+  protected val defaultExecutorCores = sparkConf.get(EXECUTOR_CORES)
 
```

Comment:
I am wondering if we want to make the locking semantics more formal in this class.
Earlier, it was volatiles and concurrent hashmap (or sets back by concurrent hashmap) to eliminate need for locking - but lot of state changes were in context of 'this' being synchronized.

Do we want to make sure all changes are guarded by a lock now ? Either use 'this' everywhere or some explicit private lock object and mark it via "\@GuardedBy"
It is becoming slightly difficult too reason about the MT-safety of this class.
Do you have any thoughts Tom ?

Reply:
sure let me take a look and see if I can simplify it.